More flexible control access for methods

trecouvr opened the pull request.

trecouvr committed:
More flexible control access for methods

apuyou commented on src/Payutc/WebApp.php:
```diff
@@ -6,6 +6,11 @@
 class WebApp {
+    
+    public static $allowed_get_methods = array(
+        'WEBSALECONFIRM::notificationPayline',
```
Pas ici !

Éventuellement dans https://github.com/payutc/server/blob/master/src/Payutc/Mapping/Services.php ou alors dans un autre Mapping. Mais pas au milieu de la WebApp :crying_cat_face:

trecouvr committed:
Move allowed get methods to mapping

trecouvr committed:
Merge branch 'master' into remove-get-access

mattgu74 commented:
ça mériterait pas une page de wiki d'explication ? :)   (et un test unitaire ou deux ?) (genre vérifier que les get sont bien bloqués sauf sur la méthode ou ils sont autorisés...)

apuyou commented:
Perso je trouve pas claire du tout la structure du truc... :confused:

trecouvr commented:
L'idée c'est que par default je voudrais que toutes les méthodes ne soient accessible que en post, si tu veux 'ouvrir' une route il faut le spécifié explicitement.
Peut être que j'ai cherché trop compliqué à vouloir tout mettre dans le même array. Un array exceptions ça serait mieux ? Le probleme de separer en deux array c'est qu'on prend le risque que les deux array ne soient pas synchros (genre tu supprimes on service et tu ne le supprimes pas dans exceptions).

trecouvr committed:
Merge pull request #306 from feuloren/master

POSS3::transaction renvoie l'id de la transaction

apuyou commented:
Ce serait carrément plus lisible avec des array séparés ! On risque de désynchroniser, mais uniquement dans le sens on oublie les exceptions, ce qui n'est pas critique vu que le service n'existe plus (et qu'on finira bien par nettoyer si ça arrive).

apuyou committed:
Merge branch 'remove-get-access' of https://github.com/trecouvr/server into remove-get-access

apuyou commented:
Le service à autoriser en GET est PAYLINE/notification, et pas WEBSALE/notificationPayline (qui est appelé par Casper avec le JSON client en POST).

Aussi, si on essaie en GET une méthode qui n'existe pas, on obtient l'erreur comme quoi GET est interdit (alors que c'est juste la méthode qui n'existe pas).

apuyou committed:
Correction de la protection des GET

- le service à autoriser est PAYLINE/notification
- un array avec les exceptions pour garder la syntaxe de l'array actuel
